lua/octo/pickers/telescope/provider.lua

```diff
@@ -767,76 +767,110 @@ end
+              teams = user.teams.nodes,
+            }
+          else
+            vim.list_extend(orgs[user.login].teams, user.teams.nodes)
+          end
+        end
+      end
+    end
+
+    local results = {}
+    -- process orgs with teams
+    for _, user in pairs(users) do
+      table.insert(results, user)
+    end
+    for _, org in pairs(orgs) do
+      org.login = string.format("%s (%d)", org.login, #org.teams)
+      table.insert(results, org)
+    end
+    return results
+  end
+end
+
+local function get_mentionable_users()
+  local repo = utils.get_remote_name()
+  local owner, name = utils.split_repo(repo)
+  local query = graphql("mentionable_users_query", owner, name, { escape = true })
+  local output = gh.run {
+    args = { "api", "graphql", "--paginate", "-f", string.format("query=%s", query) },
+    mode = "sync",
+  }
+  if not output then
+    return {}
+  end
+
+  local responses = utils.get_pages(output)
+
+  local users = {}
+
+  for _, resp in ipairs(responses) do
+    for _, user in ipairs(resp.data.repository.mentionableUsers.nodes) do
+      table.insert(users, {
+        id = user.id,
+        login = user.login,
+      })
+    end
+  end
+
+  return users
+end
+
 function M.select_user(cb)
   local opts = vim.deepcopy(dropdown_opts)
   opts.layout_config = {
     width = 0.4,
     height = 15,
   }
 
-  --local queue = {}
-  local function get_user_requester()
-    return function(prompt)
-      -- skip empty queries
-      if not prompt or prompt == "" or utils.is_blank(prompt) then
-        return {}
-      end
-      local query = graphql("users_query", prompt)
-      local output = gh.run {
-        args = { "api", "graphql", "--paginate", "-f", string.format("query=%s", query) },
-        mode = "sync",
-      }
-      if output then
-        local users = {}
-        local orgs = {}
-        local responses = utils.get_pages(output)
-        for _, resp in ipairs(responses) do
-          for _, user in ipairs(resp.data.search.nodes) do
-            if not user.teams then
-              -- regular user
-              if not vim.tbl_contains(vim.tbl_keys(users), user.login) then
-                users[user.login] = {
-                  id = user.id,
-                  login = user.login,
-                }
-              end
-            elseif user.teams and user.teams.totalCount > 0 then
-              -- organization, collect all teams
-              if not vim.tbl_contains(vim.tbl_keys(orgs), user.login) then
-                orgs[user.login] = {
-                  id = user.id,
-                  login = user.login,
-                  teams = user.teams.nodes,
-                }
-              else
-                vim.list_extend(orgs[user.login].teams, user.teams.nodes)
-              end
-            end
-          end
-        end
+  -- local finder = finders.new_dynamic {
+  --     entry_maker = entry_maker.gen_from_user(),
+  --     fn = get_user_requester(),
+  -- }
```

Comment:
TODO: Toggle with configuration option? What are your thoughts here @pwntester

Reply:
went will a key "users" which takes values "search", "mentionable", or "assignable"
Let me know your thoughts on this approach